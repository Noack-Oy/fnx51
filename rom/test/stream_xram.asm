
; ****************************
; * XRAM Stream Test Program *
; ****************************

; This program wtites and reads XRAM through a stream


; **********************
; * Header Definitions *
; **********************
.inc ../global/variables.equ
.inc ../global/sfr.equ
.inc ../serial/sfr.equ


; *************
; * Main Code *
; *************

.org 0
.inc ../global/init.inc

; *** test/stream.asm ***

	acall	serial_init

.equ	scratch,	0x180

	push	out
	push	out+1
	mov	dptr,	#stream_xram_write
	mov	out,	dpl
	mov	out+1,	dph
	mov	dptr,	#scratch
	mov	stream_out,	dpl
	mov	stream_out+1,	dph
	mov	dptr,	#test_message
	acall	print_text
	mov	r0,	stream_out
	mov	r1,	stream_out+1
	acall	print_int_u16
	pop	out+1
	pop	out

	clr	a
	mov	r3,	a
	mov	r2,	a
	mov	r1,	a
	mov	r0,	a
	mov	stream_in,	a
	mov	stream_in+1,	a
	mov	dptr,	#stream_xram_read
	mov	in,	dpl
	mov	in+1,	dph
	acall	dump
	acall	dump

	sjmp	*

test_message:
	.db	"Hello, XRAM: "
test_message_end:
	.db	0


; *********************
; * Library Functions *
; *********************
.inc ../serial/init.inc
.inc ../serial/rx.inc
.inc ../serial/tx.inc
.inc ../stream/xram_read.inc
.inc ../stream/xram_write.inc
.inc ../print/char.inc
.inc ../print/text.inc
.inc ../print/hex.inc
.inc ../print/int.inc
.inc ../read/char.inc
.inc ../util/dump.inc
.inc ../util/xch.inc
.inc ../util/regbank.inc

.equ panic_out, serial_tx
.inc ../util/panic.inc
