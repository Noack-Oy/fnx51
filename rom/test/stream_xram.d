stream_xram.lst stream_xram.hex: stream_xram.asm ../global/variables.equ ../global/sfr.equ ../serial/sfr.equ ../global/init.inc ../serial/init.inc ../serial/rx.inc ../serial/tx.inc ../stream/xram_read.inc ../stream/xram_write.inc ../print/char.inc ../print/text.inc ../print/hex.inc ../print/int.inc ../read/char.inc ../util/dump.inc ../util/xch.inc ../util/regbank.inc ../util/panic.inc
