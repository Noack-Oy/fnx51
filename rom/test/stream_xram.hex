:1000000075815090006C85823085833190007485B5
:1000100082328583331162C032C03390009F858263
:1000200032858333900180858236858337900054F2
:1000300011CBA836A9373171D033D032E4FBFAF9AD
:10004000F8F534F53590007C85823085833151D6C2
:1000500051D680FE48656C6C6F2C205852414D3A49
:100060002000759852759ABF759B1E223098FDC26C
:1000700098E599223099FDC299F59922C08EC082E7
:10008000C083E58E55E14510F58E85348285358334
:10009000E0A3858234858335D083D082D08E22C080
:1000A0008EC082C083F582E58E55E14510F58EE560
:1000B00082853682853783F0A3858236858337D063
:1000C00083D082D08E22C032C03322C0E0E8C0E0AC
:1000D000E4F893600611C608E870F7D0E0F8D0E0C5
:1000E00022C411E5C4C0E0540F2490D43440D44459
:1000F0002011C6D0E022C0E0E911E1E811E1D0E032
:1001000022C0E0EB11E1EA11E1E911E1E811E1D0EF
:10011000E022C0E0C0F030E710F5F0742D11C6E524
:10012000F0F4048004C0E0C0F088F0C0F0780075FE
:10013000F064846005243011C608E5F075F00A8487
:1001400028600598243011C6E5F0243011C6D0E0AF
:10015000F8D0F0D0E022C0E0E8C0E0E9C0E030E74D
:1001600018742D11C6E8F42401F8E9F43400F9807C
:1001700008C0E0E8C0E0E9C0E0EBC0E0C0F07A0011
:1001800075F0FFC3E89410F8E99427F905F050F4EE
:10019000E82410F8E93427F9E5F06005243011C6A9
:1001A0000A75F0FFC3E894E8F8E99403F905F05004
:1001B000F4E824E8F8E93403F9E5F02A60069A2423
:1001C0003011C60A75F0FFC3E89464F8E99400F9A9
:1001D00005F050F4E82464F8E93400F9E5F02A6009
:1001E000069A243011C60AE875F00A842A60059A36
:1001F000243011C6E5F0243011C6D0F0D0E0FAD09A
:10020000E0F9D0E0F8D0E022C0E0C0F0C082C083C6
:10021000719FE6FC08E6FD08E6FE08E6FF30E72FE2
:10022000742D11C6ECF42401FCEDF43400FDEEF461
:100230003400FEEFF43400FF8015C0E0C0F0C0824F
:10024000C083719FE6FC08E6FD08E6FE08E6FF7540
:10025000F000900286E493F8A3E493F9A3E493FA00
:10026000A3E493FBA351AA25F0600895F024300580
:10027000F011C6B80ADFEC243011C671B8D083D0B3
:1002800082D0F0D0E02200CA9A3B00E1F5058096CA
:10029000980040420F00A086010010270000E803EC
:1002A0000000640000000A000000C0F0C375F0FF09
:1002B000EC98FCED99FDEE9AFEEF9BFF05F050F0F7
:1002C000EC28FCED39FDEE3AFEEF3BFFE5F0D0F017
:1002D00022C030C03122C0F0C0E0F5F0ECC0E0ED4B
:1002E000C0E0EEC0E0EFC0E07192E581F82410F5C7
:1002F00081791051D108F6D5F003198002D9F47133
:100300009231017192743A11C6E824F029F87A10FA
:1003100008EA54037004742011C6EAD39950087493
:100320002011C611C68003E611E1742011C6DAE07F
:10033000E824F0F8742011C6747C11C67410C399B7
:10034000F9FA08E6B420004005B47F004002742E9C
:1003500011C6DAEE747C11C6740D11C6740A11C68A
:10036000E824F0F8EC29FCED3400FDEE3400FEEF5B
:100370003400FFE5F0600241F1E58124F0F5817180
:1003800092D0E0FFD0E0FED0E0FDD0E0FCD0E0D0A5
:10039000F022C8CCC8C9CDC9CACECACBCFCB22C0E7
:1003A000E0E5D05418B4180271CBE5D02408F5D09C
:1003B000541824F8F8D0E022C0E0E5D054187002B8
:1003C00071CBE5D024F8F5D0D0E022C2AF90007414
:1003D0008582328583339003E511CBD0E011E1D0E3
:1003E000E011E180FE0D0A202121212070616E695B
:0403F0006320400046
:00000001FF
