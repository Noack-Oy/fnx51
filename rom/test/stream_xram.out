BOOTLOADER
> U
< U
> :020000050B00EE
< :020000050B00EE7B.\r\n
WRITE: stream_xram.hex
> :1000000075815090006C85823085833190007485B5
< :1000000075815090006C85823085833190007485B5.\r\n
> :1000100082328583331162C032C03390009F858263
< :1000100082328583331162C032C03390009F858263.\r\n
> :1000200032858333900180858236858337900054F2
< :1000200032858333900180858236858337900054F2.\r\n
> :1000300011CBA836A9373171D033D032E4FBFAF9AD
< :1000300011CBA836A9373171D033D032E4FBFAF9AD.\r\n
> :10004000F8F534F53590007C85823085833151D6C2
< :10004000F8F534F53590007C85823085833151D6C2.\r\n
> :1000500051D680FE48656C6C6F2C205852414D3A49
< :1000500051D680FE48656C6C6F2C205852414D3A49.\r\n
> :100060002000759852759ABF759B1E223098FDC26C
< :100060002000759852759ABF759B1E223098FDC26C.\r\n
> :1000700098E599223099FDC299F59922C08EC082E7
< :1000700098E599223099FDC299F59922C08EC082E7.\r\n
> :10008000C083E58E55E14510F58E85348285358334
< :10008000C083E58E55E14510F58E85348285358334.\r\n
> :10009000E0A3858234858335D083D082D08E22C080
< :10009000E0A3858234858335D083D082D08E22C080.\r\n
> :1000A0008EC082C083F582E58E55E14510F58EE560
< :1000A0008EC082C083F582E58E55E14510F58EE560.\r\n
> :1000B00082853682853783F0A3858236858337D063
< :1000B00082853682853783F0A3858236858337D063.\r\n
> :1000C00083D082D08E22C032C03322C0E0E8C0E0AC
< :1000C00083D082D08E22C032C03322C0E0E8C0E0AC.\r\n
> :1000D000E4F893600611C608E870F7D0E0F8D0E0C5
< :1000D000E4F893600611C608E870F7D0E0F8D0E0C5.\r\n
> :1000E00022C411E5C4C0E0540F2490D43440D44459
< :1000E00022C411E5C4C0E0540F2490D43440D44459.\r\n
> :1000F0002011C6D0E022C0E0E911E1E811E1D0E032
< :1000F0002011C6D0E022C0E0E911E1E811E1D0E032.\r\n
> :1001000022C0E0EB11E1EA11E1E911E1E811E1D0EF
< :1001000022C0E0EB11E1EA11E1E911E1E811E1D0EF.\r\n
> :10011000E022C0E0C0F030E710F5F0742D11C6E524
< :10011000E022C0E0C0F030E710F5F0742D11C6E524.\r\n
> :10012000F0F4048004C0E0C0F088F0C0F0780075FE
< :10012000F0F4048004C0E0C0F088F0C0F0780075FE.\r\n
> :10013000F064846005243011C608E5F075F00A8487
< :10013000F064846005243011C608E5F075F00A8487.\r\n
> :1001400028600598243011C6E5F0243011C6D0E0AF
< :1001400028600598243011C6E5F0243011C6D0E0AF.\r\n
> :10015000F8D0F0D0E022C0E0E8C0E0E9C0E030E74D
< :10015000F8D0F0D0E022C0E0E8C0E0E9C0E030E74D.\r\n
> :1001600018742D11C6E8F42401F8E9F43400F9807C
< :1001600018742D11C6E8F42401F8E9F43400F9807C.\r\n
> :1001700008C0E0E8C0E0E9C0E0EBC0E0C0F07A0011
< :1001700008C0E0E8C0E0E9C0E0EBC0E0C0F07A0011.\r\n
> :1001800075F0FFC3E89410F8E99427F905F050F4EE
< :1001800075F0FFC3E89410F8E99427F905F050F4EE.\r\n
> :10019000E82410F8E93427F9E5F06005243011C6A9
< :10019000E82410F8E93427F9E5F06005243011C6A9.\r\n
> :1001A0000A75F0FFC3E894E8F8E99403F905F05004
< :1001A0000A75F0FFC3E894E8F8E99403F905F05004.\r\n
> :1001B000F4E824E8F8E93403F9E5F02A60069A2423
< :1001B000F4E824E8F8E93403F9E5F02A60069A2423.\r\n
> :1001C0003011C60A75F0FFC3E89464F8E99400F9A9
< :1001C0003011C60A75F0FFC3E89464F8E99400F9A9.\r\n
> :1001D00005F050F4E82464F8E93400F9E5F02A6009
< :1001D00005F050F4E82464F8E93400F9E5F02A6009.\r\n
> :1001E000069A243011C60AE875F00A842A60059A36
< :1001E000069A243011C60AE875F00A842A60059A36.\r\n
> :1001F000243011C6E5F0243011C6D0F0D0E0FAD09A
< :1001F000243011C6E5F0243011C6D0F0D0E0FAD09A.\r\n
> :10020000E0F9D0E0F8D0E022C0E0C0F0C082C083C6
< :10020000E0F9D0E0F8D0E022C0E0C0F0C082C083C6.\r\n
> :10021000719FE6FC08E6FD08E6FE08E6FF30E72FE2
< :10021000719FE6FC08E6FD08E6FE08E6FF30E72FE2.\r\n
> :10022000742D11C6ECF42401FCEDF43400FDEEF461
< :10022000742D11C6ECF42401FCEDF43400FDEEF461.\r\n
> :100230003400FEEFF43400FF8015C0E0C0F0C0824F
< :100230003400FEEFF43400FF8015C0E0C0F0C0824F.\r\n
> :10024000C083719FE6FC08E6FD08E6FE08E6FF7540
< :10024000C083719FE6FC08E6FD08E6FE08E6FF7540.\r\n
> :10025000F000900286E493F8A3E493F9A3E493FA00
< :10025000F000900286E493F8A3E493F9A3E493FA00.\r\n
> :10026000A3E493FBA351AA25F0600895F024300580
< :10026000A3E493FBA351AA25F0600895F024300580.\r\n
> :10027000F011C6B80ADFEC243011C671B8D083D0B3
< :10027000F011C6B80ADFEC243011C671B8D083D0B3.\r\n
> :1002800082D0F0D0E02200CA9A3B00E1F5058096CA
< :1002800082D0F0D0E02200CA9A3B00E1F5058096CA.\r\n
> :10029000980040420F00A086010010270000E803EC
< :10029000980040420F00A086010010270000E803EC.\r\n
> :1002A0000000640000000A000000C0F0C375F0FF09
< :1002A0000000640000000A000000C0F0C375F0FF09.\r\n
> :1002B000EC98FCED99FDEE9AFEEF9BFF05F050F0F7
< :1002B000EC98FCED99FDEE9AFEEF9BFF05F050F0F7.\r\n
> :1002C000EC28FCED39FDEE3AFEEF3BFFE5F0D0F017
< :1002C000EC28FCED39FDEE3AFEEF3BFFE5F0D0F017.\r\n
> :1002D00022C030C03122C0F0C0E0F5F0ECC0E0ED4B
< :1002D00022C030C03122C0F0C0E0F5F0ECC0E0ED4B.\r\n
> :1002E000C0E0EEC0E0EFC0E07192E581F82410F5C7
< :1002E000C0E0EEC0E0EFC0E07192E581F82410F5C7.\r\n
> :1002F00081791051D108F6D5F003198002D9F47133
< :1002F00081791051D108F6D5F003198002D9F47133.\r\n
> :100300009231017192743A11C6E824F029F87A10FA
< :100300009231017192743A11C6E824F029F87A10FA.\r\n
> :1003100008EA54037004742011C6EAD39950087493
< :1003100008EA54037004742011C6EAD39950087493.\r\n
> :100320002011C611C68003E611E1742011C6DAE07F
< :100320002011C611C68003E611E1742011C6DAE07F.\r\n
> :10033000E824F0F8742011C6747C11C67410C399B7
< :10033000E824F0F8742011C6747C11C67410C399B7.\r\n
> :10034000F9FA08E6B420004005B47F004002742E9C
< :10034000F9FA08E6B420004005B47F004002742E9C.\r\n
> :1003500011C6DAEE747C11C6740D11C6740A11C68A
< :1003500011C6DAEE747C11C6740D11C6740A11C68A.\r\n
> :10036000E824F0F8EC29FCED3400FDEE3400FEEF5B
< :10036000E824F0F8EC29FCED3400FDEE3400FEEF5B.\r\n
> :100370003400FFE5F0600241F1E58124F0F5817180
< :100370003400FFE5F0600241F1E58124F0F5817180.\r\n
> :1003800092D0E0FFD0E0FED0E0FDD0E0FCD0E0D0A5
< :1003800092D0E0FFD0E0FED0E0FDD0E0FCD0E0D0A5.\r\n
> :10039000F022C8CCC8C9CDC9CACECACBCFCB22C0E7
< :10039000F022C8CCC8C9CDC9CACECACBCFCB22C0E7.\r\n
> :1003A000E0E5D05418B4180271CBE5D02408F5D09C
< :1003A000E0E5D05418B4180271CBE5D02408F5D09C.\r\n
> :1003B000541824F8F8D0E022C0E0E5D054187002B8
< :1003B000541824F8F8D0E022C0E0E5D054187002B8.\r\n
> :1003C00071CBE5D024F8F5D0D0E022C2AF90007414
< :1003C00071CBE5D024F8F5D0D0E022C2AF90007414.\r\n
> :1003D0008582328583339003E511CBD0E011E1D0E3
< :1003D0008582328583339003E511CBD0E011E1D0E3.\r\n
> :1003E000E011E180FE0D0A202121212070616E695B
< :1003E000E011E180FE0D0A202121212070616E695B.\r\n
> :0403F0006320400046
< :0403F0006320400046.\r\n
> :00000001FF
< :00000001FF.\r\n
RESET
00000000: fa b8 00 10  8e d0 bc 00  b0 b8 00 00  8e d8 8e c0  |................|
00000010: fb be 00 7c  bf 00 06 b9  00 02 f3 a4  ea 21 06 00  |...|.........!..|
00000020: 00 be be 07  38 04 75 0b  83 c6 10 81  fe fe 07 75  |....8.u........u|
00000030: f3 eb 16 b4  02 b0 01 bb  00 7c b2 80  8a 74 01 8b  |.........|...t..|
00000040: 4c 02 cd 13  ea 00 7c 00  00 eb fe 00  00 00 00 00  |L.....|.........|
00000050: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000060: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000070: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000080: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000090: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000a0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000b0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000c0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000d0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000e0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000000f0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000100: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000110: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000120: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000130: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000140: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000150: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000160: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000170: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
00000180: 48 65 6c 6c  6f 2c 20 58  52 41 4d 3a  20 33 39 37  |Hello, XRAM: 397|
00000190: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000001a0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000001b0: 00 00 00 00  00 00 00 00  54 b3 4d 35  00 00 00 02  |........T.M5....|
000001c0: 03 00 0c fe  7f e8 80 00  00 00 00 e8  77 00 00 00  |............w...|
000001d0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000001e0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 00 00  |................|
000001f0: 00 00 00 00  00 00 00 00  00 00 00 00  00 00 55 aa  |..............U.|
